src/scripts/show_originals.js

```diff
@@ -6,36 +6,57 @@ import { keyToCss } from '../util/css_map.js';
 
 const excludeClass = 'xkit-show-originals-done';
 const hiddenClass = 'xkit-show-originals-hidden';
+const activeTimelineClass = 'xkit-show-originals-timeline';
 const lengthenedClass = 'xkit-show-originals-lengthened';
 
-const dashboardTimeline = '/v2/timeline/dashboard';
-const dashboardTimelineRegExp = new RegExp(dashboardTimeline);
-
-const lengthenTimelines = async () => {
-  const timeline =
-    document.querySelector(`[data-timeline="${dashboardTimeline}"]:not(.${excludeClass})`);
+let showOwnReblogs;
+let showReblogsWithContributedContent;
+let whitelistedUsernames;
+let runOnDashboard;
+let runOnPeepr;
+let runOnBlogSubscriptions;
 
-  if (timeline) {
-    timeline.classList.add(excludeClass);
-    const paginationCss = await keyToCss('manualPaginatorButtons');
+const lengthenTimeline = async (timeline) => {
+  const paginationCss = await keyToCss('manualPaginatorButtons');
 
-    if (!timeline.querySelector(paginationCss)) {
-      timeline.classList.add(lengthenedClass);
-    }
+  if (!timeline.querySelector(paginationCss)) {
+    timeline.classList.add(lengthenedClass);
   }
 };
 
-let showOwnReblogs;
-let showReblogsWithContributedContent;
-let whitelistedUsernames;
+const processTimelines = () => {
+  [...document.querySelectorAll(`[data-timeline]:not(.${excludeClass})`)]
+    .forEach(timeline => {
+      timeline.classList.add(excludeClass);
+
+      const onDashboard = timeline.dataset.timeline === '/v2/timeline/dashboard';
+      const isSinglePostPeepr = timeline.dataset.timeline.includes('permalink');
+      const onPeepr = timeline.closest('[role="dialog"]') !== null && isSinglePostPeepr === false;
+      const onBlogSubscriptions = timeline.dataset.timeline === '/v2/timeline' &&
+        timeline.dataset.which === 'blog_subscriptions';
```

Comment:
I would like multiple conditions to follow the philosophy of our array/object newline rules; either nothing gets its own line, or everything does

```suggestion
      const onBlogSubscriptions =
        timeline.dataset.timeline === '/v2/timeline' &&
        timeline.dataset.which === 'blog_subscriptions';
```